src/components/ScaleLattice.vue

```diff
@@ -0,0 +1,346 @@
+      return [];
+    }
+
+    const combinations = kCombinations(props.scale.intervals, 2);
+    return combinations
+      .map(([a, b]) => {
+        return {
+          distance: monzoEuclideanDistance(
+            equavePrimeIndex.value ?? -1, //adding a type check outside of this map doesn't seem to work so passing in -1
+            a.monzo.toIntegerMonzo(),
+            b.monzo.toIntegerMonzo()
+          ),
+          pair: [a.name, b.name],
+        };
+      })
+      .filter(({ distance }) => distance === 1);
+  } catch {
+    return [];
+  }
+});
+
+const svgElement = ref<HTMLElement | null>(null);
+
+const viewBox = computed(() => {
+  if (svgElement.value) {
+    return [
+      ...svgElement.value.querySelectorAll("text"),
+      ...svgElement.value.querySelectorAll("circle"),
+    ].reduce(
+      ({ minX, maxX, minY, maxY }, el) => {
+        const { x, y, width, height } = el.getBBox({
+          stroke: true,
+          fill: true,
+        });
+        return {
+          minX: Math.min(minX, x),
+          minY: Math.min(minY, y),
+          maxX: Math.max(maxX, x + width),
+          maxY: Math.max(maxY, y + height),
+        };
+      },
+      {
+        minX: Infinity,
+        minY: Infinity,
+        maxX: -Infinity,
+        maxY: -Infinity,
+      }
+    );
+  }
+  return {
+    minX: 0,
+    minY: 0,
+    maxX: 0,
+    maxY: 0,
+  };
+});
+
+function isNodeErrorMessage(nodeValue: any): nodeValue is string {
+  return typeof nodeValue === "string";
+}
+
+const error = computed(() => {
+  if (!canLatticePrimes.value) {
+    return "Can only make lattice of JI scales up to 23-limit";
+  }
+  if (isNodeErrorMessage(nodes.value)) {
+    return (
+      nodes.value ||
+      "There was an unknown error while making this lattice, please try again or with another scale."
+    );
+  }
+  return null;
+});
+</script>
+
+<template>
+  <div ref="container" class="container">
+    <div v-if="error">
+      <p>{{ error }}</p>
+    </div>
+    <div v-else>
+      <svg
+        v-if="Array.isArray(nodes)"
+        ref="svgElement"
+        id="lattice"
+        class="lattice"
+        xmlns="http://www.w3.org/2000/svg"
+        preserveAspectRatio="xMidYMid meet"
+        style="background-color: transparent"
+        :viewBox="
+          [
+            viewBox.minX,
+            viewBox.minY,
+            viewBox.maxX - viewBox.minX,
+            viewBox.maxY - viewBox.minY,
+          ].join(' ')
+        "
+      >
+        <line
+          v-for="edge of edges"
+          class="edge"
+          :key="edge.pair.join()"
+          :x1="nodesByLabel[edge.pair[0]].x"
+          :x2="nodesByLabel[edge.pair[1]].x"
+          :y1="nodesByLabel[edge.pair[0]].y"
+          :y2="nodesByLabel[edge.pair[1]].y"
+        />
+        <circle
+          v-for="(node, index) of nodes"
+          :key="index"
+          :cx="node.x"
+          :cy="node.y"
+          r="2.5"
+          :class="`node   ${
```

Comment:
Vue supports dictionaries as classes so this reads better as something like `{node: true, heldDegreee: heldScaleDegrees.has(node.index)}`.